docs/fidelity-bonds.md

```diff
@@ -113,8 +147,109 @@ coins using `python3 wallet-tool.py <walletname> -m 0 freeze`.
 Once unfrozen and untimelocked the coins can be spent normally with the scripts
 `sendpayment.py`, `tumber.py`, or yield generator.
 
+### How many coins to lock up and for how long?
+
+Fidelity bonds UTXOs are valuable as soon as they confirmed. The simplified formula for a fidelity
+bond value with locked coins is:
+
+    bond_value = (locked_coins * exp(interest_rate * locktime))^2
```

Comment:
```suggestion
    bond_value = (locked_coins * exp(min(interest_rate * locktime, locked_coins)))^2
```

I think that's an important part for deciding how many or how long to lock funds.

Reply:
That formula is wrong, I guess you meant to write `(locked_coins * max(1, exp(interest_rate * locktime)))^2` ?

The effect of having a ceiling at the burned coin value is described in the text. It's not the only feature I missed out in this simplification (e.g. the bond value decays after the locktime passes), the full formula is here: https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/872/commits/272e2926c8d24806fb415d05bacf4e754cbf01e6#diff-b48a1717de3d71415880b8a18d7e461225bd8ddfdc09010254439897bc19fd6fR2403

I think the value of simplifying is worth missing out on some features. I could put the word "simplified" in bold to make it even clearer that people need to read more?

Reply:
I was referring to point three of the list, specifically the sentence

    […]the sacrificed value is capped at the value of the burned coins.[…]

maybe I put the `min()` at the wrong place?

Reply:
Did you read the gist on the formulas? https://gist.github.com/chris-belcher/87ebbcbb639686057a389acb9ab3e25b (I think you saw it already, but it does try to justify the formulas there).

Your proposal can't be right on first principles, because you're putting a coin value in the exponent not a time value (e^(interest-rate*time) is an expression representing exponential decay or growth of a value over time, like Ae^(kx) in general), whereas e^(coin value) is not logical in that context (value grows as a function of time, nothing is growing exponentially as a function of monetary amount itself).

Having said that I'm still trying to grok every aspect of the reasoning here myself, so I won't comment further for now :)

Reply:
> That formula is wrong, I guess you meant to write (locked_coins * max(1, exp(interest_rate * locktime)))^2 ?

Ah, yes. After reading the gist I understand now. `interest_rate * locktime < 1` holds for most cases, which was not immediately obvious to me from the explanation in the doc. That made the `exp()` a lot more confusing than it's supposed to be. `locktime` is supposed to be in years, isn't it?

Reply:
`Locktime` can be in any unit (e.g. years) because `interest_rate` is in the inverse unit (something per year). The argument to `exp` is dimensionless.

Reply:
Oh derp, been too long since I did any Physics :) Yes exponents are always dimensionless because you can't multiply a number by itself "3 metres times" :) and interest rates are always T^-1 because they're a ratio (dimensionless) per unit time.